presto-parquet/src/test/java/io/prestosql/parquet/TestTupleDomainParquetPredicate.java

```diff
@@ -352,15 +356,27 @@ public void testDate()
                 .withMessage("Corrupted statistics for column \"DateColumn\" in Parquet file \"testFile\": [min: 200, max: 100, num_nulls: 0]");
     }
 
-    @Test
-    public void testTimestampMillis()
+    @DataProvider
+    public Object[][] timestampPrecision()
+    {
+        int nanoOfSecond = 123456789;
+        Instant baseTime = Instant.from(LocalDateTime.of(1970, 1, 19, 10, 28, 52, nanoOfSecond).atZone(ZoneOffset.UTC));
```

Comment:
Would it make sense to use LocalDateTime here?

Reply:
I think we're dealing with timestamp as local date/time values (year,month,day,hour,minute,second).
Therefore LDT better represents this, than an "Instant in UTC".

(I guess this is the same reason you added `DecodedTimestamp` class instead of (ab)using `Instant`)